lib/ChangeDueDateDialog/ChangeDueDate.js

```diff
@@ -0,0 +1,294 @@
+          });
+        } else {
+          this.props.onDueDateChangeFailed(error);
+        }
+      });
+  }
+
+  getUpdatedLoans() {
+    const { loanSelection, datetime } = this.state;
+    const selectedLoanIds = Object.keys(pickBy(loanSelection, value => value));
+
+    return this.props.loans
+      .filter(loan => selectedLoanIds.includes(loan.id))
+      .map(loan => ({
+        ...loan,
+        dueDate: datetime,
+        action: 'dueDateChanged',
+      }));
+  }
+
+  validateLoanChange(loan) {
+    return loan;
+  }
+
+  performLoanChange(loan) {
+    if (!loan) return null;
+
+    this.props.mutator.loanId.replace(loan.id);
+    return this.props.mutator.loan.PUT(loan);
+  }
+
+  toggleLoanSelection(loan) {
+    const { loanSelection } = this.state;
+
+    this.setState({
+      loanSelection: {
+        ...loanSelection,
+        [loan.id]: !(loanSelection[loan.id])
+      }
+    });
+  }
+
+  toggleBulkLoanSelection() {
+    const { loanSelection } = this.state;
+
+    const someLoansUnselected = Object.values(loanSelection).includes(false);
+    Object.keys(loanSelection).forEach(key => { loanSelection[key] = someLoansUnselected; });
+
+    this.setState({ loanSelection });
+  }
+
+  render() {
+    const { loans } = this.props;
+
+    return (
+      <div>
+        { loans.length > 1 ?
+          <p>
+            <SafeHTMLMessage
+              id="stripes-smart-components.cddd.itemsSelected"
+              values={{ count: loans.length }}
+            />
+            {this.state.warnings.map((warning, i) => (
+              <span key={i} style={{ color: 'orange' }}>
+                <Icon size="medium" icon="validation-error" color="orange" />
+                {warning}
+              </span>
+            ))}
+          </p>
+          : null
+        }
+        <DueDatePicker
+          stripes={this.props.stripes}
+          onChange={this.handleDateTimeChanged}
+          initialValues={{
+            date: this.getLastDueDate(),
+            time: ChangeDueDate.DEFAULT_TIME,
+          }}
+          dateProps={{ label: 'Date*' }}
+          timeProps={{ label: 'Time *' }}
+        />
+        <this.connectedLoanList
+          stripes={this.props.stripes}
+          alerts={this.props.alerts}
+          loans={loans}
+          loanSelection={this.state.loanSelection}
+          onToggleLoanSelection={this.toggleLoanSelection}
+          onToggleBulkLoanSelection={this.toggleBulkLoanSelection}
+        />
+        <Layout className="textRight">
+          <Button onClick={this.props.onCancel}>
+            <FormattedMessage id="stripes-core.button.cancel" />
+          </Button>
+          <Button buttonStyle="primary" onClick={this.changeDueDate}>
+            <FormattedMessage id="stripes-core.button.saveAndClose" />
+          </Button>
+        </Layout>
+      </div>
+    );
+  }
+}
+
+export default reduxForm({ form: 'cddd' })(ChangeDueDate);
```

Comment:
@doytch I'm probably missing something here but do we need `reduxForm` here?

Reply:
Nope we don't. Refactoring artifact I missed.